src/Graph/DOT.elm

```diff
@@ -0,0 +1,172 @@
+module Graph.DOT
+    exposing
+        ( Rankdir(..)
+        , Styles
+        , defaultStyles
+        , output
+        , outputWithStyles
+        , outputWithStylesAndAttributes
+        )
+
+{-| This module provides a means of converting the `Graph` data type into a
+valid [DOT](https://en.wikipedia.org/wiki/DOT_(graph_description_language))
+string for visualizing your graph structure.
+
+You can easily preview your graph by inserting the generated string into an
+online GraphViz tool like <https://dreampuf.github.io/GraphvizOnline/>.
+
+You can also dynamically draw your graph in your application by sending the
+string over a port to the javascript version of the GraphViz library,
+<https://github.com/mdaines/viz.js/> (see the examples there fore more
+specifics on how to embed the generated visualization).
+
+@docs output
+
+
+# Attrs
+
+GraphViz allows for customizing the graph's look via "Attrs."
+
+@docs Styles, Rankdir, defaultStyles, outputWithStyles, outputWithStylesAndAttributes
+
+-}
+
+import Dict exposing (Dict)
+import Graph exposing (Edge, Graph, Node, edges, get, nodes)
+
+
+{-| Converts a `Graph` into a valid DOT string.
+Note that you must supply conversion functions for node labels and edge labels to strings.
```

Comment:
Maybe add a sentence about what happens when the function for node/edge returns Nothing.
It will still be rendered in the graphviz output, but with empty attributes list. In such cases graphviz uses node ID as node label.